VisualStudio 2015 - Compile errors

After some time, related to my closed issue #40, install VisualStudio 2015, download code, unzip, open project file "plc-utils-2015.sln" and clisk "Build solution", got lot of errors. Content of log window here:

1>------ Build started: Project: lib_slac, Configuration: Release Win32 ------
2>------ Build started: Project: lib_ether, Configuration: Release Win32 ------
3>------ Build started: Project: lib_pib, Configuration: Release Win32 ------
4>------ Build started: Project: lib_plc, Configuration: Release Win32 ------
2>  channel.c
4>  Antiphon.c
3>  pibfile.c
2>  closechannel.c
3>  pibfile1.c
4>  Attributes.c
2>  fcs.c
3>  pibfile2.c
4>  Attributes1.c
3>  piblock.c
1>  evse_cm_atten_char.c
1>  evse_cm_mnbc_sound.c
1>  evse_cm_set_key.c
4>  Attributes2.c
1>  evse_cm_slac_match.c
2>  gethwaddr.c
2>  getifname.c
3>  pibpeek1.c
3>  pibpeek2.c
3>  pibscalers.c
3>  psread.c
1>  evse_cm_slac_param.c
4>  BootDevice.c
1>  evse_cm_start_atten_char.c
2>  openchannel.c
4>  BootDevice1.c
1>  pev_cm_atten_char.c
2>  readpacket.c
1>  pev_cm_mnbc_sound.c
2>  sendpacket.c
3>  ruledump.c
4>  BootDevice2.c
4>  BootFirmware1.c
4>  BootFirmware2.c
4>  BootParameters1.c
1>  pev_cm_set_key.c
1>  pev_cm_slac_match.c
1>  pev_cm_slac_param.c
1>  pev_cm_start_atten_char.c
4>  BootParameters2.c
1>  slac_connect.c
4>  ChangeIdent.c
2>  lib_ether.vcxproj -> C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_ether.lib
3>  lib_pib.vcxproj -> C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_pib.lib
5>------ Build started: Project: lib_tools, Configuration: Release Win32 ------
6>------ Build started: Project: lib_nvm, Configuration: Release Win32 ------
6>  fdmanifest.c
6>  manifest.c
6>  nvm.c
6>  nvmfile.c
6>  nvmfile1.c
4>  chipset.c
6>  nvmfile2.c
6>  nvmlock2.c
6>  nvmpeek.c
4>  Confirm.c
4>  DeviceIdent.c
6>  nvmpeek1.c
6>  nvmpeek2.c
4>  Devices.c
4>  Display.c
6>  nvmseek1.c
4>  EmulateHost.c
6>  nvmseek2.c
4>  EmulateHost64.c
4>  ExecuteApplets.c
4>  ExecuteApplets1.c
4>  ExecuteApplets2.c
4>  FactoryDefaults.c
4>  FactoryReset.c
6>  lib_nvm.vcxproj -> C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_nvm.lib
7>------ Build started: Project: lib_key, Configuration: Release Win32 ------
4>  Failure.c
4>  FlashDevice1.c
7>  HPAVKeyDAK.c
7>  HPAVKeyNID.c
7>  HPAVKeyNMK.c
7>  HPAVKeyOut.c
7>  HPAVKeySHA.c
7>  HPAVKeySpec.c
7>  keys.c
7>  MACPasswords.c
7>  putpwd.c
7>  RNDPasswords.c
4>  FlashDevice2.c
7>  SHA256Block.c
4>  FlashFirmware.c
7>  SHA256Fetch.c
7>  SHA256Ident.c
4>  FlashMOD.c
4>  FlashNVM.c
7>  SHA256Match.c
7>  SHA256Print.c
7>  SHA256Reset.c
7>  SHA256Write.c
7>  strnpwd.c
4>  FlashParameters.c
4>  FlashSoftloader.c
7>  lib_key.vcxproj -> C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_key.lib
8>------ Build started: Project: lib_mme, Configuration: Release Win32 ------
5>  assist.c
5>  b64dump.c
5>  basespec.c
5>  binout.c
5>  bytespec.c
1>  slac_debug.c
5>  checkfilename.c
5>  checksum32.c
5>  chrout.c
1>  slac_session.c
5>  clr32bitmap.c
5>  codelist.c
1>  slac_structs.c
5>  codename.c
5>  config.c
5>  dataspec.c
5>..\..\tools\checksum32.c(48): error C2036: 'const void *': unknown size
5>  debug.c
5>  decdecode.c
5>  decout.c
5>  decstring.c
5>  efreopen.c
5>  emalloc.c
5>  error.c
5>  extra.c
5>  fdchecksum32.c
5>  getargv.c
5>  getoptv.c
5>  gettimeofday.c
5>  hexdecode.c
5>  hexdump.c
5>  hexencode.c
5>  hexload.c
5>  hexoffset.c
5>  hexout.c
5>  hexpeek.c
5>  hexstring.c
5>  hexview.c
5>  hexwrite.c
5>  ipv4spec.c
5>  ipv6spec.c
5>  lookup.c
5>  memdecr.c
5>  memencode.c
5>  memincr.c
5>  memout.c
5>  memswap.c
5>  output.c
5>  putoptv.c
5>  regview32.c
5>  reverse.c
5>  reword.c
5>  set32bitmap.c
5>  strdecr.c
5>  strfbits.c
5>  strincr.c
5>  synonym.c
5>  todigit.c
5>  typelist.c
5>  typename.c
5>  uintspec.c
5>  version.c
5>  __bswap.c
4>  GetProperty.c
8>  ARPCPeek.c
4>  HostActionIndicate.c
8>  ARPCPrint.c
4>  HostActionResponse.c
8>  ARPCWrite.c
4>  Identity.c
8>  EthernetHeader.c
4>  Identity1.c
8>  FirmwareMessage.c
4>  Identity2.c
8>  HomePlugHeader.c
4>  InitDevice.c
4>  InitDevice1.c
4>  InitDevice2.c
8>  HomePlugHeader1.c
4>  LinkStatistics.c
8>  MMECode.c
4>  LinkStatus.c
8>  MMEMode.c
4>  ListLocalDevices.c
8>  MMEName.c
4>  ListRemoteDevices.c
8>  MMEPeek.c
4>  ListRemoteDevices1.c
8>  QualcommHeader.c
4>  ListRemoteDevices2.c
8>  QualcommHeader1.c
4>  LocalDeviceList.c
8>  readmessage.c
8>  sendmessage.c
4>  LocalDevices.c
8>  UnwantedMessage.c
1>  lib_slac.vcxproj -> C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_slac.lib
9>------ Build started: Project: lib_ram, Configuration: Release Win32 ------
10>------ Build started: Project: lib_serial, Configuration: Release Win32 ------
9>  nvram.c
9>  nvrampeek.c
9>  sdramfile.c
9>  sdramfileA.c
10>  closeport.c
10>  openport.c
10>  serial.c
4>  MakeRule.c
9>  sdrampeek.c
9>  sdramtext.c
10>..\..\serial\openport.c(102): warning C4133: 'function': incompatible types - from 'const char *' to 'LPCWSTR'
4>  MDUTrafficStats.c
4>  ModuleCommit.c
4>  ModuleDump.c
4>  ModuleRead.c
8>  lib_mme.vcxproj -> C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_mme.lib
11>------ Build started: Project: lib_nodes, Configuration: Release Win32 ------
4>  ModuleSession.c
11>  xmlattribute.c
11>  xmldata.c
11>  xmledit.c
11>  xmlelement.c
10>  lib_serial.vcxproj -> C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_serial.lib
12>------ Build started: Project: rkey, Configuration: Release Win32 ------
11>  xmlfree.c
11>  xmlnode.c
4>  ModuleSpec.c
11>  xmlopen.c
9>  lib_ram.vcxproj -> C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_ram.lib
12>  rkey.c
13>------ Build started: Project: setpib, Configuration: Release Win32 ------
11>  xmlread.c
4>  ModuleWrite.c
4>  NetInfo.c
13>  setpib.c
11>..\..\nodes\xmlread.c(39): warning C4013: 'lstat' undefined; assuming extern returning int
11>  xmlscan.c
11>  xmlschema.c
11>  xmlselect.c
12>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
14>------ Build started: Project: nvmsplit, Configuration: Release Win32 ------
13>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
15>------ Build started: Project: nvmmerge, Configuration: Release Win32 ------
4>  NetInfo1.c
11>  xmltree.c
14>  nvmsplit.c
4>  NetInfo2.c
4>  NetworkDevices.c
4>  NetworkDevices1.c
11>  xmlvalue.c
15>  nvmmerge.c
14>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
16>------ Build started: Project: weeder, Configuration: Release Win32 ------
11>  lib_nodes.vcxproj -> C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_nodes.lib
17>------ Build started: Project: psnotch, Configuration: Release Win32 ------
15>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
18>------ Build started: Project: getpib, Configuration: Release Win32 ------
17>  psnotch.c
18>  getpib.c
16>  weeder.c
4>  NetworkDevices2.c
4>  NetworkInformation.c
4>  NetworkInformation1.c
4>  NetworkInformation2.c
4>  NetworkTraffic.c
16>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
19>------ Build started: Project: int6kuart, Configuration: Release Win32 ------
17>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
20>------ Build started: Project: chknvm2, Configuration: Release Win32 ------
4>  NetworkTraffic1.c
4>  NetworkTraffic2.c
19>  int6kuart.c
18>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
21>------ Build started: Project: chkpib2, Configuration: Release Win32 ------
21>  chkpib2.c
4>  NVMSelect.c
20>  chknvm2.c
4>  NVRAMInfo.c
4>  ParseRule.c
4>  PhyRates.c
19>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
20>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
22>------ Build started: Project: chkpib, Configuration: Release Win32 ------
23>------ Build started: Project: int6kbaud, Configuration: Release Win32 ------
21>..\..\pib\chkpib2.c(193): warning C4267: '=': conversion from 'size_t' to 'uint16_t', possible loss of data
23>  int6kbaud.c
4>  PhyRates1.c
22>  fdmanifest.c
22>  chkpib.c
21>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
4>  PhyRates2.c
24>------ Build started: Project: mme, Configuration: Release Win32 ------
4>  Platform.c
4>  PLCNetworkInfo.c
23>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
4>  PLCPhyRates.c
24>  mme.c
25>------ Build started: Project: int6kdetect, Configuration: Release Win32 ------
4>  PLCReadParameterBlock.c
4>  PLCSelect.c
22>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
26>------ Build started: Project: edsu, Configuration: Release Win32 ------
25>  int6kdetect.c
4>  PLCTopologyPrint.c
4>  PrintRule.c
26>  edsu.c
4>  PushButton.c
4>  ReadFirmware.c
4>  ReadFirmware1.c
4>  ReadFirmware2.c
24>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
27>------ Build started: Project: edru, Configuration: Release Win32 ------
25>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
28>------ Build started: Project: efbu, Configuration: Release Win32 ------
4>  ReadFMI.c
4>  ReadMFG.c
4>  ReadMME.c
26>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
29>------ Build started: Project: mdiodump, Configuration: Release Win32 ------
27>  edru.c
4>  ReadNVM.c
29>  mdiodump.c
4>  ReadParameterBlock.c
4>  ReadParameters.c
4>  ReadParameters1.c
29>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
27>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
30>------ Build started: Project: mdioblock, Configuration: Release Win32 ------
31>------ Build started: Project: mdioblock2, Configuration: Release Win32 ------
31>  mdioblock2.c
30>  mdioblock.c
4>  ReadParameters2.c
28>  efbu.c
4>  ReadRules.c
4>  RemoteHosts.c
4>  Request.c
4>  ResetAndWait.c
28>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
31>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
32>------ Build started: Project: hpav, Configuration: Release Win32 ------
33>------ Build started: Project: pibcomp, Configuration: Release Win32 ------
4>  ResetDevice.c
30>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
34>------ Build started: Project: pibdump, Configuration: Release Win32 ------
4>  rules.c
34>  pibdump.c
4>  RxRates1.c
32>  hpav.c
4>  RxRates2.c
4>  SDRAMInfo.c
32>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
35>------ Build started: Project: pib2xml, Configuration: Release Win32 ------
4>  SendMME.c
34>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
36>------ Build started: Project: xml2pib, Configuration: Release Win32 ------
37>------ Build started: Project: efru, Configuration: Release Win32 ------
35>  pib2xml.c
4>  SetNMK.c
4>  SetProperty.c
4>  SignalToNoise1.c
36>  xml2pib.c
37>  efru.c
4>  SignalToNoise2.c
4>  SlaveMembership.c
4>  StartDevice1.c
35>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
38>------ Build started: Project: efeu, Configuration: Release Win32 ------
36>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
39>------ Build started: Project: modpib, Configuration: Release Win32 ------
4>  StartFirmware.c
4>  StartFirmware1.c
4>  StartFirmware2.c
38>  efeu.c
37>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
39>  modpib.c
40>------ Build started: Project: chknvm, Configuration: Release Win32 ------
4>  StationRole.c
40>  chknvm.c
4>  ToneMaps1.c
38>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
41>------ Build started: Project: efsu, Configuration: Release Win32 ------
4>  ToneMaps2.c
4>  Topology.c
39>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
41>  efsu.c
42>------ Build started: Project: config2cfg, Configuration: Release Win32 ------
4>  Topology1.c
40>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
43>------ Build started: Project: pcapdevs, Configuration: Release Win32 ------
4>  Topology2.c
42>  config2cfg.c
4>  Traffic.c
43>  pcapdevs.c
4>  Traffic1.c
4>  Traffic2.c
41>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
44>------ Build started: Project: hpavkeys, Configuration: Release Win32 ------
4>  Transmit.c
42>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
4>  UpgradeDevice1.c
45>------ Build started: Project: sdram, Configuration: Release Win32 ------
45>  sdram.c
43>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
46>------ Build started: Project: hpavkey, Configuration: Release Win32 ------
4>  VersionInfo1.c
4>  VersionInfo2.c
44>  hpavkeys.c
4>  WaitForReset.c
4>  WaitForRestart.c
44>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
46>  hpavkey.c
47>------ Build started: Project: ptsctl, Configuration: Release Win32 ------
4>  WaitForStart.c
4>  WatchdogReport.c
4>  WriteCFG.c
45>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
46>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
48>------ Build started: Project: mac2pw, Configuration: Release Win32 ------
49>------ Build started: Project: mac2pwd, Configuration: Release Win32 ------
49>  mac2pwd.c
50>------ Build started: Project: mdiogen, Configuration: Release Win32 ------
49>..\..\key\mac2pwd.c(233): warning C4244: '=': conversion from 'uint64_t' to 'unsigned int', possible loss of data
49>..\..\key\mac2pwd.c(239): warning C4244: '=': conversion from 'uint64_t' to 'unsigned int', possible loss of data
4>..\..\plc\WaitForStart.c(143): warning C4267: '=': conversion from 'size_t' to 'uint8_t', possible loss of data
48>  mac2pw.c
4>  WriteExecuteApplet2.c
48>..\..\key\mac2pw.c(224): warning C4244: '=': conversion from 'uint64_t' to 'unsigned int', possible loss of data
4>  WriteExecuteFirmware.c
48>..\..\key\mac2pw.c(230): warning C4244: '=': conversion from 'uint64_t' to 'unsigned int', possible loss of data
4>  WriteExecuteFirmware1.c
48>..\..\key\mac2pw.c(236): warning C4244: '=': conversion from 'uint64_t' to 'unsigned int', possible loss of data
4>  WriteExecuteFirmware2.c
50>  mdiogen.c
4>  WriteExecuteParameters.c
48>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
51>------ Build started: Project: psgraph, Configuration: Release Win32 ------
4>  WriteExecuteParameters1.c
49>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
52>------ Build started: Project: psin, Configuration: Release Win32 ------
4>  WriteExecuteParameters2.c
51>  psgraph.c
4>  WriteExecutePIB.c
52>  psin.c
4>  WriteFirmware.c
4>  WriteFirmware1.c
4>  WriteFirmware2.c
50>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
51>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
53>------ Build started: Project: pskey, Configuration: Release Win32 ------
54>------ Build started: Project: psout, Configuration: Release Win32 ------
53>  pskey.c
54>  psout.c
52>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
4>  WriteMEM.c
4>  WriteMOD.c
4>  WriteNVM.c
4>  WriteParameters.c
4>  WriteParameters1.c
54>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
4>  WriteParameters2.c
53>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
4>  WritePIB.c
4>  lib_plc.vcxproj -> C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_plc.lib
55>------ Build started: Project: int6krate, Configuration: Release Win32 ------
56>------ Build started: Project: int6kwait, Configuration: Release Win32 ------
57>------ Build started: Project: sada, Configuration: Release Win32 ------
58>------ Build started: Project: int6kstat, Configuration: Release Win32 ------
56>  int6kwait.c
57>  sada.c
58>  int6kstat.c
55>  int6krate.c
56>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
55>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
57>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
58>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
59>------ Build started: Project: int6krule, Configuration: Release Win32 ------
60>------ Build started: Project: CMEncrypt, Configuration: Release Win32 ------
61>------ Build started: Project: int6ktest, Configuration: Release Win32 ------
62>------ Build started: Project: int6ktone, Configuration: Release Win32 ------
59>  int6krule.c
61>  int6ktest.c
60>  CMEncrypt.c
62>  int6ktone.c
59>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
62>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
63>------ Build started: Project: int6kboot, Configuration: Release Win32 ------
64>------ Build started: Project: int6keth, Configuration: Release Win32 ------
61>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
60>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
65>------ Build started: Project: int6klog, Configuration: Release Win32 ------
64>  int6keth.c
63>  int6kboot.c
65>  int6klog.c
66>------ Build started: Project: int64host, Configuration: Release Win32 ------
64>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
63>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
67>------ Build started: Project: int6kmod, Configuration: Release Win32 ------
66>  int64host.c
65>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
68>------ Build started: Project: pibrump, Configuration: Release Win32 ------
68>  pibrump.c
66>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
69>------ Build started: Project: pibruin, Configuration: Release Win32 ------
70>------ Build started: Project: int6klist, Configuration: Release Win32 ------
67>  int6kmod.c
67>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
71>------ Build started: Project: amptest, Configuration: Release Win32 ------
69>  pibruin.c
70>  int6klist.c
68>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
72>------ Build started: Project: amptone, Configuration: Release Win32 ------
71>  amptest.c
72>  amptone.c
69>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
70>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
73>------ Build started: Project: amptool, Configuration: Release Win32 ------
74>------ Build started: Project: amprate, Configuration: Release Win32 ------
71>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
73>  amptool.c
74>  amprate.c
75>------ Build started: Project: ampboot, Configuration: Release Win32 ------
72>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
75>  ampboot.c
76>------ Build started: Project: amphost, Configuration: Release Win32 ------
76>  amphost.c
73>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
74>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
77>------ Build started: Project: amplist, Configuration: Release Win32 ------
78>------ Build started: Project: ampstat, Configuration: Release Win32 ------
75>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
79>------ Build started: Project: int6kmdio, Configuration: Release Win32 ------
77>  amplist.c
78>  ampstat.c
79>  int6kmdio.c
76>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
80>------ Build started: Project: int6kmdio2, Configuration: Release Win32 ------
80>  int6kmdio2.c
77>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
81>------ Build started: Project: mdustats, Configuration: Release Win32 ------
78>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
82>------ Build started: Project: plcotst, Configuration: Release Win32 ------
79>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
83>------ Build started: Project: plcget, Configuration: Release Win32 ------
81>  mdustats.c
82>  plcotst.c
83>  plcget.c
80>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
84>------ Build started: Project: plcset, Configuration: Release Win32 ------
84>  plcset.c
81>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
85>------ Build started: Project: plcfwd, Configuration: Release Win32 ------
82>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
86>------ Build started: Project: plcrate, Configuration: Release Win32 ------
85>  plcfwd.c
83>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
87>------ Build started: Project: plctest, Configuration: Release Win32 ------
86>  plcrate.c
87>  plctest.c
84>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
88>------ Build started: Project: plctool, Configuration: Release Win32 ------
88>  plctool.c
85>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
86>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
89>------ Build started: Project: plcboot, Configuration: Release Win32 ------
90>------ Build started: Project: plchost, Configuration: Release Win32 ------
89>  plcboot.c
90>  plchost.c
87>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
91>------ Build started: Project: plcstat, Configuration: Release Win32 ------
88>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
92>------ Build started: Project: plcmdio32, Configuration: Release Win32 ------
91>  plcstat.c
92>  plcmdio32.c
89>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
93>------ Build started: Project: plcmdio16, Configuration: Release Win32 ------
90>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
94>------ Build started: Project: plclist, Configuration: Release Win32 ------
94>  plclist.c
93>  plcmdio16.c
91>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
95>------ Build started: Project: plcwait, Configuration: Release Win32 ------
92>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
96>------ Build started: Project: ampwait, Configuration: Release Win32 ------
95>  plcwait.c
96>  ampwait.c
93>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
94>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
97>------ Build started: Project: plctone, Configuration: Release Win32 ------
98>------ Build started: Project: ampID, Configuration: Release Win32 ------
98>  ampID.c
96>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
95>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
99>------ Build started: Project: plcrule, Configuration: Release Win32 ------
100>------ Build started: Project: plcID, Configuration: Release Win32 ------
100>  plcID.c
101>------ Build started: Project: int6k, Configuration: Release Win32 ------
98>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
102>------ Build started: Project: int6kid, Configuration: Release Win32 ------
101>  int6k.c
102>  int6kid.c
100>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
103>------ Build started: Project: amprule, Configuration: Release Win32 ------
104>------ Build started: Project: coqos_add, Configuration: Release Win32 ------
103>  amprule.c
104>  coqos_add.c
101>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
102>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
105>------ Build started: Project: coqos_info, Configuration: Release Win32 ------
106>------ Build started: Project: coqos_man, Configuration: Release Win32 ------
106>  coqos_man.c
105>  coqos_info.c
103>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
107>------ Build started: Project: coqos_rel, Configuration: Release Win32 ------
104>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
108>------ Build started: Project: evse, Configuration: Release Win32 ------
107>  coqos_rel.c
108>  evse.c
106>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
105>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
109>------ Build started: Project: int6kf, Configuration: Release Win32 ------
110>------ Build started: Project: int6khost, Configuration: Release Win32 ------
109>  int6kf.c
110>  int6khost.c
107>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
111>------ Build started: Project: pev, Configuration: Release Win32 ------
108>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
112>------ Build started: Project: plclog, Configuration: Release Win32 ------
111>  pev.c
112>  plclog.c
109>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
110>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
113>------ Build started: Project: coqos_mod, Configuration: Release Win32 ------
113>  coqos_mod.c
111>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
112>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
113>LINK : fatal error LNK1181: cannot open input file 'C:\Users\LAZNA\Desktop\open-plc-utils-master\VisualStudioNET\Release\lib_tools.lib'
========== Build: 10 succeeded, 103 failed, 0 up-to-date, 0 skipped ==========

--- a/tools/checksum32.c
+++ b/tools/checksum32.c
@@ -31,10 +31,11 @@
 
 #include "../tools/memory.h"
 
-uint32_t checksum32 (void const * memory, size_t extent, uint32_t checksum)
+uint32_t checksum32 (void const * vmemory, size_t extent, uint32_t checksum)
 
 {
 	uint32_t temp;
+	const uint8_t *memory = vmemory;
 
 	while (extent >= sizeof (checksum))
 	{

